### PR TITLE
refactor: use `token` pkg from `fabric8-common`

### LINKED ISSUE #6: Move token manager to fabric8-common
We need to update https://github.com/fabric8-services/fabric8-common/tree/master/token package and reuse it in fabric8-cluster.

The current token package in fabric8-cluster is in better state for common. So we should base token package in common on https://github.com/fabric8-services/fabric8-cluster/tree/master/token

Plus add DefaultManager() function from https://github.com/fabric8-services/fabric8-auth/blob/master/token/token.go#L62 too.

### PR DESCRIPTION
**Pull request by xcoulon**
fixes #6

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: goamiddleware/jwt_token_context.go [hunk 1]
```diff
@@ -6,8 +6,8 @@ import (
 	"net/http"
 	"strings"
 
-	"github.com/fabric8-services/fabric8-cluster/token"
 	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/fabric8-services/fabric8-common/token"
```
Comment (nurali-techie) [inline goamiddleware/jwt_token_context.go]: I was expecting that this file will not be there at all in each_service .. rather each_service would be using the jwt_token_context_middleware from https://github.com/fabric8-services/fabric8-common/blob/master/goamiddleware/jwt_token_context.go

Comment (nurali-techie) [reply goamiddleware/jwt_token_context.go]: this middleware in f8-cluster [link](https://github.com/fabric8-services/fabric8-cluster/blob/master/goamiddleware/jwt_token_context.go#L20) and f8-common [link](https://github.com/fabric8-services/fabric8-common/blob/master/goamiddleware/jwt_token_context.go#L22) are similar and meant for same purpose I guess.  In that case, we should remove this file from f8-cluster [link](https://github.com/fabric8-services/fabric8-cluster/blob/master/goamiddleware/jwt_token_context.go) and directly use middleware from f8-common.

File: Gopkg.toml [hunk 2]
```diff
@@ -98,4 +98,4 @@ required = [
 
 [[constraint]]
   name = "github.com/fabric8-services/fabric8-common"
-  revision = "51e44b47e514721ddaa3fca99e5fb5bc1a6e7eb2"
```
Comment (alexeykazakov) [inline Gopkg.toml]: Why removing revision?

File: goamiddleware/jwt_token_context.go (hunk 1 above)
Comment (alexeykazakov) [reply goamiddleware/jwt_token_context.go]: The current middleware in common was taken from WIT. It's outdated and has some issues. We should really move the middleware from cluster/auth to common and re-use it here.
With modifications suggested by Nurali - https://chat.openshift.io/developers/pl/awuewy3jubde78nw4q94oqf7gc

File: main.go [hunk 3]
```diff
@@ -16,8 +16,8 @@ import (
 	"github.com/fabric8-services/fabric8-cluster/jsonapi"
 	"github.com/fabric8-services/fabric8-cluster/migration"
 	"github.com/fabric8-services/fabric8-cluster/sentry"
-	"github.com/fabric8-services/fabric8-cluster/token"
 	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/fabric8-services/fabric8-common/token"
```
Comment (alexeykazakov) [inline main.go]: We should also initialize the default token manager - https://github.com/fabric8-services/fabric8-auth/blob/master/main.go#L151

File: goamiddleware/jwt_token_context.go (hunk 1 above)
Comment (alexeykazakov) [reply goamiddleware/jwt_token_context.go]: See https://github.com/fabric8-services/fabric8-common/pull/27

File: main.go (hunk 3 above)
Comment (alexeykazakov) [reply main.go]: Done in https://github.com/fabric8-services/fabric8-cluster/pull/14/commits/561b4c9b7a2cfabb78d0d720274b455bd0f5b30d

**Pull request merged**